packages/edit-site/src/components/template-switcher/index.js

```diff
@@ -70,52 +74,82 @@ export default function TemplateSwitcher( {
 		setThemePreviewVisible( () => false );
 	};
 
+	const [ homeId, setHomeId ] = useState();
+
+	useEffect( () => {
+		const effect = async () => {
+			try {
+				const { success, data } = await fetch(
+					addQueryArgs( '/', { '_wp-find-template': true } )
+				).then( ( res ) => res.json() );
+				if ( success ) {
+					let newHomeId = data.ID;
+					if ( newHomeId === null ) {
+						const { getEntityRecords } = resolveSelect( 'core' );
+						newHomeId = (
+							await getEntityRecords( 'postType', 'wp_template', {
+								resolved: true,
+								slug: data.post_name,
+							} )
+						 )[ 0 ].id;
+					}
+					setHomeId( newHomeId );
+				} else {
+					throw new Error();
+				}
+			} catch ( err ) {
+				setHomeId( null );
+			}
+		};
+		effect();
+	}, [] );
+
 	const { currentTheme, template, templateParts } = useSelect(
-		( select ) => {
-			const { getCurrentTheme, getEntityRecord } = select( 'core' );
-			const _template = getEntityRecord(
-				'postType',
-				'wp_template',
-				activeId
-			);
+		( _select ) => {
+			const {
+				getCurrentTheme,
+				getEntityRecord,
+				getEntityRecords,
+			} = _select( 'core' );
+			const theme = getCurrentTheme();
+
 			return {
-				currentTheme: getCurrentTheme(),
-				template: {
-					label: _template ? (
-						<TemplateLabel
-							template={ _template }
-							homeId={ homeId }
-						/>
-					) : (
-						__( 'Loading…' )
-					),
-					value: activeId,
-					slug: _template ? _template.slug : __( 'Loading…' ),
-					content: _template?.content,
-				},
-				templateParts: templatePartIds.map( ( id ) => {
-					const templatePart = getEntityRecord(
-						'postType',
-						'wp_template_part',
-						id
-					);
-					return {
-						label: templatePart ? (
-							<TemplateLabel template={ templatePart } />
-						) : (
-							__( 'Loading…' )
-						),
-						value: id,
-						slug: templatePart
-							? templatePart.slug
-							: __( 'Loading…' ),
-					};
-				} ),
+				currentTheme: theme,
+				template: getEntityRecord(
+					'postType',
+					'wp_template',
+					activeId
+				),
+				templateParts: theme
+					? getEntityRecords( 'postType', 'wp_template_part', {
+							resolved: true,
+							theme: theme.stylesheet,
+					  } )
+					: null,
```

Comment:
Shouldn't we request all the template parts in the current template instead? That would make more sense from the user's perspective.

Reply:
Makes sense! Fixed by 2f9c4393f5.

Should I remove the `theme` arg now? :thinking: Looks like we don't need it anymore.

Reply:
Yes, unless you see a clear use-case for it.

Reply:
A list of all existing template parts for the current theme might be relevant for a template inserter, see e.g. https://github.com/WordPress/gutenberg/pull/22612/files#r435884349.

I'll leave it there.